Fig.Core/IniFileSettingsSource.cs

```diff
@@ -18,11 +18,11 @@ public IniFileSettingsSource(string path)
         {
             _path = path;
         }
-        
+
         internal static IEnumerable<(string, string)> Parse(IEnumerable<string> lines)
         {
-            var sectionMatcher = new Regex(@"^\[\s*(?<section>[A-Z0-9:]+)\s*\]$", RegexOptions.IgnoreCase);
-            var keyValueMatcher = new Regex(@"^(?<key>[A-Z0-9:]+)\s*=\s*(?<value>.*)$", RegexOptions.IgnoreCase);
+            var sectionMatcher = new Regex(@"^\[(.+)\]$", RegexOptions.IgnoreCase);
+            var keyValueMatcher = new Regex(@"^\s*([^#].+?)\s*=\s*(.*)$", RegexOptions.IgnoreCase);
```

Comment:
These are bit too "open" in the sense of which keys and sections are valid. For example '[' and ']', tabs and spaces are valid in a section. But maybe we should define what the valid syntax is and come back to this later. There might even exist a formal INI file definition somewhere which we ought to follow or at least document on how we differ.